File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums2) {
        int cnt = 0, n = nums2.size();
        vector<int> nums;
        nums.push_back(nums2[0]);

        for(int i = 1 ; i < n ; i++) {
            if(nums2[i] != nums2[i-1]) nums.push_back(nums2[i]);
        }

        // for(int i: nums) cout<<i<<" ";

        for(int i = 1;  i < nums.size() - 1; i++) {
            // valley
            if(nums[i - 1] >= nums[i] && nums[i] < nums[i+1]) cnt++;
            // hill
            else if(nums[i-1] < nums[i] && nums[i] >= nums[i+1]) cnt++;
        }
        // for(int i = 1; i < n - 1 ; i++) {
            // // valley
            // if(nums[i - 1] >= nums[i] && nums[i] < nums[i+1]) cnt++;
            // // hill
            // else if(nums[i-1] < nums[i] && nums[i] >= nums[i+1]) cnt++;
        // }
        return cnt;
    }
};

File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        cnt = 0
        uniq = [nums[0]]

        for i in range(1, len(nums)) :
            if nums[i - 1] != nums[i] : 
                uniq.append(nums[i])

        print(uniq)

        for i in range(1, len(uniq) - 1) : 
            if uniq[i - 1] > uniq[i] and uniq[i] < uniq[i+1] : cnt += 1
            if uniq[i - 1] < uniq[i] and uniq[i] > uniq[i+1] : cnt += 1

        return cnt